packages/sirv/index.js

```diff
@@ -69,59 +83,84 @@ function send(req, res, file, stats, headers={}) {
 	fs.createReadStream(file, opts).pipe(res);
 }
 
-module.exports = function (dir, opts={}) {
+function isEncoding(name, type, headers) {
+	headers['Content-Encoding'] = type;
+	headers['Content-Type'] = mime.getType(name.replace(/\.([^.]*)$/, '')) || '';
+}
+
+function toHeaders(name, stats, isEtag) {
+	let headers = {
+		'Content-Length': stats.size,
+		'Content-Type': mime.getType(name) || '',
+		'Last-Modified': stats.mtime.toUTCString(),
+	};
+	if (isEtag) headers['ETag'] = `W/"${stats.size}-${stats.mtime.getTime()}"`;
+	if (/\.br$/.test(name)) isEncoding(name, 'brotli', headers);
+	if (/\.gz$/.test(name)) isEncoding(name, 'gzip', headers);
+	return headers;
+}
+
+export default function (dir, opts={}) {
 	dir = resolve(dir || '.');
 
 	let isNotFound = opts.onNoMatch || is404;
-	let extensions = opts.extensions || ['html', 'htm'];
 	let setHeaders = opts.setHeaders || noop;
 
-	if (opts.dev) {
-		return function (req, res, next) {
-			let stats, file, uri=decodeURIComponent(req.path || req.pathname || parser(req).pathname);
-			let arr = [uri].concat(toAssume(uri, extensions)).map(x => normalize(join(dir, x))).filter(x => {
-				return x.startsWith(dir) && fs.existsSync(x);
-			});
-
-			while (file = arr.shift()) {
-				stats = fs.statSync(file);
-				if (stats.isDirectory()) continue;
-				setHeaders(res, uri, stats);
-				return send(req, res, file, stats, {
-					'Content-Type': mime.getType(file),
-					'Last-Modified': stats.mtime.toUTCString(),
-					'Content-Length': stats.size,
-				});
-			}
-			return next ? next() : isNotFound(req, res);
-		}
+	let extensions = opts.extensions || ['html', 'htm'];
+	let gzips = opts.gzip && extensions.map(x => `${x}.gz`).concat('gz');
+	let brots = opts.brotli && extensions.map(x => `${x}.br`).concat('br');
```

Comment:
Rollback please decodeURIComponent because this issue is back: https://github.com/lukeed/sirv/pull/21